developers/documentation/docs_template.md

```diff
@@ -0,0 +1,121 @@
+
+This utility allows you to pass an active Viewer as a parameter and produces a screenshot of the Viewer at that 
+point in time. This screenshot will be displayed to the user in the notebook.
+
+```{code-cell} ipython3
+import napari
+from napari.utils import nbscreenshot
+
+viewer = napari.Viewer()
+# opens sample data and adds layer to the Viewer
+viewer.open_sample('scikit-image', 'cells3d')
+
+# takes a screenshot and produces it as output for this cell
+nbscreenshot(viewer)
+```
+
+### 2. Hide input cells
+
+As you can see, it's simple to produce screenshots of the napari Viewer in your notebooks. However, if you look through napari's
+existing documentation, none of the code cells include calls to `nbscreenshot`, yet the screenshots are still produced. In fact,
+it would be distracting if all the code cells included `nbscreenshot`, and might be frustrating for users who
+want to execute these notebooks in their own workflows.
+
+To avoid this frustration, we place calls to `nbscreenshot` in a hidden cell in your notebooks.
+You can completely remove input (i.e. the code that's running) in a notebook cell by adding a `remove-input` tag to the cell metadata.
+
+How you add cell tags depends on how you're editing your notebook. 
+
+1. If you're working in Jupyter notebook,
+you can open up the Tags toolbar for your cell using `View -> Cell Toolbar -> Tags`. You can then add any tags you want
+(e.g. `remove-input`) by typing into the text entry box of the toolbar and clicking `Add Tag`. 
+Here's what the Tags toolbar looks like.
+
+![Jupyter notebook cell with Tags toolbar highlighted by a black square in the top right of the cell. Tags toolbar involves a button
+with ellipses for seeing existing tags, a text entry box for adding new tags, and an Add Tag button](assets/jupyter_cell_tags.png)
+
+2. If you're editing a MyST Markdown file directly, you can add tags to your code blocks like so:
+
+    ```{code-cell}
+    :tags: [remove-input]
+
+    print("Your code here")
+    ```
+
+#### What to put in hidden cells
+
+Alongside your call to `nbscreenshot`, you can also place other potentially distracting code in these tagged cells, 
+such as resizing the Viewer window or opening a menu. In general, if you're running code the reader isn't meant to run,
+this should be in a hidden cell.
+The screenshot below is produced by the following code, which has been hidden from you using the `remove-input` tag.
+
+```python
+from napari.utils import nbscreenshot
+
+viewer.window._qt_window.resize(750, 550)
+nbscreenshot(viewer)
+```
+
+Note how we've included the `nbscreenshot` import in this hidden cell. Even though in the
+example above we imported `nbscreenshot` to show its functionality, you should place the
+import in a hidden cell when you write your documentation.
+
+```{code-cell} ipython3
+:tags: [remove-input]
+
+from napari.utils import nbscreenshot
+
+viewer.window._qt_window.resize(750, 550)
+nbscreenshot(viewer)
+```
```

Comment:
Should we have a short list of common functions that might get used in hidden cells? I'm thinking very small, just the list & links to those functions in the documentation, and only for the types of things people are likely to adjust without thinking about in the GUI and might not know how to do programmatically.

These might include:
- resizing a window (like you have already mentioned here)
- Toggling the layer visibility (eg: several layers are displayed, but we want only one to show up in the screenshot)
- Setting the dims slider position
- Adjusting the contrast
- Switching the viewer mode (2D -> 3D, or gridview)
- ...anything else I'm forgetting?

Reply:
I've added a list of these settings. Where I could find the property/function in our API reference I've linked it. Where I couldn't, I've provided a code snippet to show how the setting could be used.